pallets/confidential-asset/src/lib.rs

```diff
@@ -76,48 +71,35 @@ pub struct MercatAccount {
     pub encryption_pub_key: EncryptionPubKey,
 }
 
+type Identity<T> = identity::Module<T>;
+
 decl_storage! {
     trait Store for Module<T: Trait> as ConfidentialAsset {
 
         /// Contains the mercat accounts for an identity.
         pub MercatAccounts get(fn mercat_account):
-            double_map hasher(twox_64_concat) IdentityId,
-            hasher(blake2_128_concat) EncryptedAssetIdWrapper
-            => MercatAccount;
+              double_map hasher(twox_64_concat) IdentityId,
+              hasher(blake2_128_concat) EncryptedAssetIdWrapper
+              => MercatAccount;
 
         /// Contains the encrypted balance of a mercat account.
         pub MercatAccountBalance get(fn mercat_account_balance):
-            double_map hasher(twox_64_concat) IdentityId,
-            hasher(blake2_128_concat) EncryptedAssetIdWrapper
-            => EncryptedBalanceWrapper;
-
-        /// Contains the list of all valid ticker names.
-        /// The process around creating and storing this data will likely change as a result of
-        /// CRYP-153.
-        pub ValidAssetIds get(fn valid_asset_ids): Vec<AssetId>;
+              double_map hasher(twox_64_concat) IdentityId,
+              hasher(blake2_128_concat) EncryptedAssetIdWrapper
+              => EncryptedBalanceWrapper;
+
+        /// List of Tickers of the type ConfidentialAsset.
+        /// () -> List of confidential tickers
+        pub ConfidentialTickers get(fn confidential_tickers): Vec<AssetId>;
```

Comment:
Any access would require decoding the entire `Vec`. There is a way around that, just use a characteristic function instead - a map from `AssetId` to `bool`.

`BTreeSet` would be better than `Vec`: `log(n)` reads as opposed to `n`. But again it would require reading the whole `BTreeSet` from runtime storage, which can be avoided.

Reply:
@aafshar-poly & I had a discussion about this couple of weeks ago. Arash do you guys still need the whole vector of asset_ids to validate the proof? I think it was more related to the index of asset_id. we can re-discuss this.

Reply:
Yes, we need the whole AssetID vector. The list of all confidential asset IDs ever created to be exact.

Reply:
OK, got you. The only place I found in in this PR where `confidential_tickers` is called is an application of `AccountCreator::create`. It takes a reference to the `Vec`. I would suggest modifying `AccountCreator::create` to take `&[Scalar]` instead for uniformity and also because slices are better as arguments as opposed to `&Vec<_>`.